File: backend/spec/factories/task.rb
FactoryBot.define do
  factory :task do
    title   { "Test Task" }
    content { "Some content for the task." }
    status  { :pending }
    due_at  { nil }

    trait :in_progress do
      status { :in_progress }
    end

    trait :blocked do
      status { :blocked }
    end

    trait :completed do
      status { :completed }
    end
  end
end

File: frontend/src/types/task.ts
export type TaskStatus = 'pending' | 'in_progress' | 'blocked' | 'completed';

export interface Task {
  id: number;
  title: string;
  content: string | null;
  status: TaskStatus;
  due_at: string | null;
}

export type TaskFormValues = {
  title: string;
  content: string | null;
  status: TaskStatus;
  due_at: string | null;
};
